crates/ruma-client-api/src/filter/get_filter.rs

```diff
@@ -63,12 +63,13 @@ pub mod v3 {
         fn deserialize_response() {
             use ruma_common::api::IncomingResponse;
 
-            assert_matches!(
+            let filter = assert_matches!(
                 super::Response::try_from_http_response(
                     http::Response::builder().body(b"{}" as &[u8]).unwrap(),
                 ),
-                Ok(super::Response { filter }) if filter.is_empty()
+                Ok(super::Response { filter }) => filter
             );
+            assert!(filter.is_empty());
```

Comment:
This is probably a `vec!`? Maybe we should do `assert_matches!(filter.as_slice(), []);` to print its value if it's non-empty.

Reply:
It's not, it's a struct.